sshd-core/src/main/java/org/apache/sshd/common/kex/extension/DefaultClientKexExtensionHandler.java

```diff
@@ -133,4 +151,39 @@ protected void handleServerSignatureAlgorithms(Session session, Collection<Strin
             session.setSignatureFactories(clientAlgorithms);
         }
     }
+
+    @Override
+    public void sendKexExtensions(Session session, KexPhase phase) throws Exception {
```

Comment:
This must send something only if the server had announced ext-info-s. Otherwise the client must not send any extensions. See the logic in the DefaultServerKexExtensionHandler.